sdk/maps/Azure.Maps.Route/samples/RouteDirectionsSamples.md

```diff
@@ -0,0 +1,196 @@
+```
+
+## Asynchronous Route Direction Batch Request
+
+If there are more then `100` route direction queries, one can use asynchronous route direction batch request and wait for the request to finish later. The maximum matrix available queries are `700`.
+
+```C# Snippet:AsyncRequestRouteDirectionsBatch
+// Create a list of route direction queries
+IList<RouteDirectionQuery> queries = new List<RouteDirectionQuery>();
+
+queries.Add(new RouteDirectionQuery(
+    new List<GeoPosition>() {
+        new GeoPosition(123.751, 45.9375),
+        new GeoPosition(123.791, 45.96875),
+        new GeoPosition(123.767, 45.90625)
+    },
+    new RouteDirectionOptions()
+    {
+        TravelMode = TravelMode.Bicycle,
+        RouteType = RouteType.Economy,
+        UseTrafficData = false,
+    })
+);
+queries.Add(new RouteDirectionQuery(new List<GeoPosition>() { new GeoPosition(123.751, 45.9375), new GeoPosition(123.767, 45.90625) }));
+
+// Invoke asynchronous route direction batch request, we can get the result later via assigning `WaitUntil.Started`
+var operation = await client.RequestRouteDirectionsBatchAsync(WaitUntil.Started, queries);
+
+// After a while, get the result back
+var result = operation.WaitForCompletion();
+```
+
+The asynchronous route direction result will be cached for 14 days. User can fetch the result from server via a `RequestRouteDirectionsOperation` with the same `Id`:
+
+```C# Snippet:AsyncRequestRouteDirectionsBatchWithOperationId
+// Create a list of route direction queries
+IList<RouteDirectionQuery> queries = new List<RouteDirectionQuery>();
+
+queries.Add(new RouteDirectionQuery(
+    new List<GeoPosition>() {
+        new GeoPosition(123.751, 45.9375),
+        new GeoPosition(123.791, 45.96875),
+        new GeoPosition(123.767, 45.90625)
+    },
+    new RouteDirectionOptions()
+    {
+        TravelMode = TravelMode.Bicycle,
+        RouteType = RouteType.Economy,
+        UseTrafficData = false,
+    })
+);
+queries.Add(new RouteDirectionQuery(new List<GeoPosition>() { new GeoPosition(123.751, 45.9375), new GeoPosition(123.767, 45.90625) }));
+
+// Invoke asynchronous route direction batch request
+var operation = client.RequestRouteDirectionsBatch(WaitUntil.Started, queries);
+
+// Get the operation ID and store somewhere
+var operationId = operation.Id;
+```
+
+Within 14 days, user can use the same operation ID to fetch the same result. One precondition is the client endpoint should be the same:
+
+```C# Snippet:AsyncRequestRouteDirectionsBatchWithOperationId2
+// Within 14 days, users can retrive the cached result with operation ID
+// The `endpoint` argument in `client` should be the same!
+var newRouteDirectionOperation = new RequestRouteDirectionsOperation(client, operationId);
+var result = newRouteDirectionOperation.WaitForCompletion();
+```
+
+## Route Direction Result
+
+Route direction result is stored in `RouteDirectionsBatchResult` type. User can access the `BatchItems` from `RouteDirectionsBatchResult`:
```

Comment:
```suggestion
Route direction result is stored in `RouteDirectionsBatchResult` type. You can access the `BatchItems` from `RouteDirectionsBatchResult`:
```
The term "user" in the context of samples would generally refer to the developer reading it.  Rather than being in the third person voice, it reads more friendly to use "you" and avoid terms like "user" or "the developer".